File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsVolumeView.java
package com.sequenceiq.cloudbreak.cloud.aws.view;

import com.sequenceiq.cloudbreak.cloud.model.Volume;

public class AwsVolumeView {

    private static final char[] DEVICE_CHAR = {'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k'};

    private Volume volume;

    private String device;

    private int index;

    public AwsVolumeView(Volume volume, int index) {
        this.volume = volume;
        this.device = "" + DEVICE_CHAR[index];
    }

    public int getIndex() {
        return index;
    }

    public String getMount() {
        return volume.getMount();
    }

    public String getType() {
        return volume.getType();
    }

    public int getSize() {
        return volume.getSize();
    }

    public String getDevice() {
        return device;
    }
}

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsVolumeView.kt
package com.sequenceiq.cloudbreak.cloud.aws.view

import com.sequenceiq.cloudbreak.cloud.model.Volume

class AwsVolumeView(private val volume: Volume, index: Int) {

    val device: String

    val index: Int = 0

    init {
        this.device = "" + DEVICE_CHAR[index]
    }

    val mount: String
        get() = volume.mount

    val type: String
        get() = volume.type

    val size: Int
        get() = volume.size

    companion object {

        private val DEVICE_CHAR = charArrayOf('b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k')
    }
}